### PR TITLE
Add @octokit/webhooks

### PR DESCRIPTION
**Pull request by tyankatsu0105**
re https://github.com/actions/toolkit/issues/147

- Refactor packages/github's interface
  - Quote from @octokit/webhooks declaration

### CODE REVIEW

**Comment by ericsciple** [comment]
@hross?

**Comment by ericsciple** [comment]
@hross sorry at a glance i thought this was related to the change you made. it looks like this is about the github.context object instead

**Comment by hross** [comment]
@ericsciple these changes are in the same spirit as my changes (importing types instead of re-defining our own). I think they are good changes that we should add but I don't have a good way of testing them. @thboop has a similar issue on his plate. Maybe worth setting up something in canary that would test the toolkit with webhooks like we did for my changes.

**Comment by srb2** [comment]
thank you very much :) appreciated

**Comment by thboop** [comment]
Hey @srb2, thanks for the excellent PR! 

@hross, @ericsciple  I'm going to merge this unless you have any reservations, I've tested the changes they seem good and its the correct step forward for the project.

File: packages/github/src/interfaces.ts [hunk 1]
```diff
@@ -1,39 +1,15 @@
...
-  }
-  pull_request?: {
-    [key: string]: any
-    number: number
-    html_url?: string
-    body?: string
-  }
+  repository?: Webhooks.PayloadRepository
+  issue?: Webhooks.WebhookPayloadIssuesIssue
+  pull_request?: Webhooks.WebhookPayloadPullRequest
```
Comment (thboop) [inline packages/github/src/interfaces.ts]: This should be WebhookPayloadPullRequestPullRequest

**Comment by tyankatsu0105** [comment]
Umm...
This PR's author is me. @thboop

File: packages/github/src/interfaces.ts (hunk 1 above)
Comment (tyankatsu0105) [reply packages/github/src/interfaces.ts]: Thanks.
Done.

Comment (tyankatsu0105) [reply packages/github/src/interfaces.ts]: There is type error

Comment (tyankatsu0105) [reply packages/github/src/interfaces.ts]: Fix.

**Comment by tyankatsu0105** [comment]
@thboop 
Bumping to v3.0.0 is great.
But I need more quote type from @octokit/webhooks.
Because I think there are some types that not support yet.
So, I suppose it's early timing when bumping version.
What do you think about this?

**Comment by ericsciple** [comment]
@bryanmacfarlane thoughts?

Do you know what the original motivation was, that these types were redefined rather than re-exported?

The size concerns me. Installing `@octokit/webhooks@^7` adds 350k uncompressed, 50k compressed

There are thousands of lines of potential [WebhookPayload* type definitions](https://github.com/octokit/webhooks.js/blob/master/index.d.ts). I'm wondering about the original motivation of `@actions/github`. Why only a subset of types? Why not OR all types together? Are all types are even compatible (can they be OR'd)?

Assuming only a subset is desired (say for size consideration), at some point the solution is for the consumer to import `@octokit/webhooks` and cast to the correct type. Should we update scenario docs?

Is the motivation for bumping to v3.0.0 because `PayloadRepository` was removed? Do we need to make this breaking change? An alternative would be to re-export the interface:
```
import {PayloadRepository} from '@octokit/webhooks'
export {PayloadRepository}
```

**Comment by thboop** [comment]
@tyankatsu0105 ,

I've added #310 which provides instructions on how to do this manually if needed. 

If there is a particularly field you think would be helpful to add, we can add that to the existing typescript definitions, feel free to open a PR or create an issue!

Currently, this pr sets each field to its most common definition. For example, it sets `pull_request` to `WebhookPayloadPullRequestPullRequest`. That holds true for the `pull_request` event, but the `pull_request_review` event has pull_request correspond to the type `WebhookPayloadPullRequestReviewPullRequest`. These have different fields, so the type information is inaccurate for this event.

If we are going to to import the `@octokit/webhook` definitions, we should look at importing the full payload types. The WebhookPayload field here is really just a union type between all of the possible payloads.
```
  | Webhooks.WebhookPayloadPush // If the event is "push"
  | Webhooks.WebhookPayloadPullRequest // if the event is "pull_request"
  | Webhooks.WebhookPayloadPullRequestReview // if the event is "pull_request_review"
 // ect
```

This would require bumping the major version of the module, and we don't want to push out a major version for just this at this time. I've tagged #147 with the enhancement flag and we will consider implementing it a future release.

**Comment by tyankatsu0105** [comment]
@thboop 
OK. I understand.
I close this PR.

**Pull request closed without merging**